### PR TITLE
document CodegenClass

### PR DESCRIPTION
**Pull request by fredemmott**

### CODE REVIEW

File: src/CodegenClass.php [hunk 1]
```diff
@@ -33,56 +35,76 @@
...
     return $this;
   }
 
+  /** @selfdocumenting */
   public function setIsAbstract(bool $value = true): this {
     $this->isAbstract = $value;
     return $this;
   }
 
+  /** Set the parent class of the generated class. */
```
Comment (JoelMarcey) [inline src/CodegenClass.php]: Interesting that this wasn't called `setParent`

File: src/CodegenClass.php [hunk 2]
```diff
@@ -13,15 +13,17 @@
...
  *
- * codegen_class('Foo')
+ * To construct an instance, use `ICodegenFactory::codegenClass()`.
+ *
+ * ```
+ * $factory->codegenClass('Foo')
  *  ->setExtends('bar')
  *  ->setIsFinal()
  *  ->addMethod(codegen_method('foobar'))
  *  ->render();
```
Comment (JoelMarcey) [inline src/CodegenClass.php]: Where is this method defined?

File: src/CodegenClass.php (hunk 1 above)
Comment (fredemmott) [reply src/CodegenClass.php]: I assumed appendBodyToBuilder was from an interface, but it isn't

I'm only saying 'parent' here to not re-use 'extends' - it's extends to match the keyword in generated code.

File: src/CodegenClass.php (hunk 2 above)
Comment (fredemmott) [reply src/CodegenClass.php]: the `HackBuilderRenderer` trait

**Pull request merged**